packages/skill-api/src/core/services/stripe-checkout.ts

```diff
@@ -155,22 +177,21 @@ export async function stripeCheckout({
           coupon: coupon.id,
         })
       } else if (merchantCoupon && merchantCoupon.identifier) {
-        // no ppp for bulk purchases
-        const isNotPPP = merchantCoupon.type !== 'ppp'
-        if (isNotPPP || quantity === 1) {
-          const {id} = await stripe.promotionCodes.create({
-            coupon: merchantCoupon.identifier,
-            max_redemptions: 1,
-            expires_at: TWELVE_FOUR_HOURS_FROM_NOW,
-          })
-          discounts.push({
-            promotion_code: id,
-          })
-        }
+        const {id} = await stripe.promotionCodes.create({
+          coupon: merchantCoupon.identifier,
+          max_redemptions: 1,
+          expires_at: TWELVE_HOURS_FROM_NOW,
+        })
+        discounts.push({
+          promotion_code: id,
+        })
       }
 
+      // TODO: Could we throw this error much earlier in the function and
+      // avoid generating promo codes and making other queries when we are
+      // eventually going to throw anyway?
       if (!loadedProduct) {
-        throw new CheckoutError('No product was found', productId as string)
+        throw new CheckoutError('No product was found', productId)
       }
```

Comment:
I think it would make sense to push this all up to where `loadedProduct` gets queried so that we can circumvent a bunch of logic and queries that are going to get thrown away anyway.

Thoughts?